### PR TITLE
Add config for browser, using configs for eslint-plugin-react, -import, -promise

### PR DESCRIPTION
**Pull request by davidtheclark**
I pieced together a draft of an ESLint config for (most of our) browser code. That meant adding configs related to React and ES2015 modules, and I threw in some eslint-plugin-promise rules, as well, since I just found out about that plugin and those 3 rules seem nice.

I restructured the code a little bit with the thought that this organization could facilitate selective composition of configs. For example, once some folks start using ES2015 modules in Node, they could also start using the `import.js` config (if eslint-plugin-node doesn't take over that responsibility). If we don't want to add the `promise.js` config to the main `node.js` one, any Node author who does want to use it could still do so (e.g. `extends: require.resolve('@mapbox/eslint-config-mapbox/promise')`). According to the idea that rules for a given plugin are captured in one config named for the plugin, I deleted `lib/base-node.js` and put those rules in `node.js`. `base.js` itself is out in the main directory in case anybody wants to just use that; and each other config extends `base.js`.

I'm curious if you think this is a good approach or not. 

If so, I will follow up by double-checking that all this works as expected; looking for feedback from other frontend devs on the rules I've chosen; probably adding another config to accommodate Prettier usage; and modifying the docs.

### CODE REVIEW

**Comment by davidtheclark** [comment]
BTW: It's a real bummer that we cannot install these ESLint plugins as dependencies of the shareable config package, making the multi-plugin config easier to consume. So I was thinking about making a separate package that encapsulates all the required dependencies you'd need for the browser config, and exposes a thin wrapper CLI with the commands you'd use (along the lines of [XO](https://github.com/xojs/xo)) — e.g. instead of `eslint .` you'd do something like `mbx-enforcer lint .`.

**Comment by rclark** [comment]
Thanks for this @davidtheclark. Could you make some changes to the `readme.md`:

- each of the different options that the module exposes should be documented
- examples of `extends` values for each different option
- list additional plugins that have to be installed depending on the option you use

**Comment by davidtheclark** [comment]
@rclark what do you think of the documentation I added?

File: readme.md [hunk 1]
```diff
@@ -41,3 +43,46 @@
...
+- [`@mapbox/eslint-config-mapbox`**`/import`**](./import.js)
+  - Depends on [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import).
+  - For ES2015 modules (`import` and `export`).
+- [`@mapbox/eslint-config-mapbox`**`/promise`**](./promise.js)
+  - Depends on [eslint-plugin-promise](https://github.com/xjamundx/eslint-plugin-promise).
+  - For `Promise`s.
+
+### With Prettier
+
+If you are using [Prettier](https://prettier.io/) to format your JS, add [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) at the end of your `"extends"` array. It will turn off all ESLint rules related to the code-style preferences that Prettier already addresses.
```
Comment (rclark) [inline readme.md]: Should also mention that `eslint-config-prettier` needs to be installed.

**Comment by davidtheclark** [comment]
🆗 🆒 , made the change.

**Pull request merged**